snapshot/localmounter_freebsd.go

```diff
@@ -0,0 +1,56 @@
+// +build freebsd
+
+package snapshot
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/containerd/containerd/mount"
+	"github.com/pkg/errors"
+)
+
+func (lm *localMounter) Mount() (string, error) {
```

Comment:
Why is this copy needed? And why is it missing the single layer direct mount that exists in linux?

Reply:
>why is it missing the single layer direct mount that exists in linux?

Bind mounts are not supported in FreeBSD. Rather, we use [nullfs](https://www.freebsd.org/cgi/man.cgi?query=nullfs&apropos=0&sektion=0&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html) instead. So the condition like `(lm.mounts[0].Type == "bind" || lm.mounts[0].Type == "rbind")` must return false.


Perhaps I am missing something, but the rest of the function should handle the single layer direct mount case.


But I don't think that the copy itself is necessary, thanks!

Reply:
I don't know exactly what is the difference between nullfs and bind but if there is no technical limitation then the freebsd code should detect the single layer as well, eg. `lm.mounts[0].Type == "nullfs"`. For better performance and to avoid privileged call in tests when it is not needed. Plus to just be consistent with linux.

Reply:
Gotcha, thanks.

We still need root privileges to mount nullfs, though